core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -854,6 +854,33 @@ class ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>,
 namespace Kokkos {
 namespace Impl {
 
+template <typename T>
+struct alignas(T) volatile_wrapper {
+  T t;
+
```

Comment:
Consider adding copy ctor and operator=, or see what doesn't compile when they're `=delete;`

Reply:
Found at least one site that was doing direct assignment, so I implemented them.